src/test/java/org/opensearch/knn/index/VectorDataTypeIT.java

```diff
@@ -425,6 +425,51 @@ public void testKNNScriptScoreWithInvalidByteQueryVector() throws Exception {
         );
     }
 
+    @SneakyThrows
+    public void testSearchWithInvalidSearchVectorType() {
+        createKnnIndexMappingWithLuceneEngine(2, SpaceType.L2, VectorDataType.FLOAT.getValue());
+        ingestL2FloatTestData();
+        Request request = new Request("POST", String.format("/%s/_search", INDEX_NAME));
+        request.setJsonEntity(
```

Comment:
Can we use `XContentBuilder` to construct json entity?

Reply:
Since it's just a simple JSON object, I think it's easier to view in current format.

Reply:
The problem with plain json string is that it look ugly in java code with all those escape character `\` and `+` sign which makes it easier to make a mistake and harder to edit.

Reply:
Updated to use `XContentBuilder `